src/pretix_ldap/ldap_connector.py

```diff
@@ -75,15 +78,15 @@ def form_authenticate(self, request, form_data):
             # wrong password
             return None
         try:
-            user = User.objects.get(email=email)
-            if user.auth_backend == self.identifier:
-                return user
-            else:
-                # user already registered with different backend
-                return None
-        except User.DoesNotExist:
-            # user does not exist yet -> create new
-            user = User(email=email)
-            user.auth_backend = self.identifier
-            user.save()
+            user = User.objects.get_or_create_for_backend(
+                self.identifier,
+                uuid,
+                email
+            )
+        except EmailAddressTakenError:
+            messages.error(
+                request, _('We cannot create your user account as a user account in this system '
```

Comment:
I don't know how I feel about this error message, since it leaks information about the registered users. Maybe just log the error and show a more generic message?

Reply:
It most likely only leaks information to someone who controls exactly that email address, unless the LDAP contains unverified email addresses

Reply:
It can be used find out if a user is registered at a specific pretix instance, can't it? I.e. I can start to enumerate email addresses and find out for which of them an account exists.

Reply:
Yeah but for every enumeration step you need to do a valid ldap login for the address you are trying to enumerate if I'm not mistaken

Reply:
Yes, the error message would only appear if you have a valid login for the LDAP, i.e. an enumeration of already "used" email addresses would be possible but only if you already have working LDAP credentials.

Reply:
I have removed for now though, so feel free to handle this as you think is best.

Reply:
Sorry, I didn't thoroughly look at this before writing that comment. Given the constraint, that you already have to have a valid login, I think the more explicit error message is fine.